### PR TITLE
Add Path.flatMapOperations to make editing Path more convenient

### PR DESCRIPTION
**Pull request by RafalSumislawski**

### CODE REVIEW

File: swagger/src/main/scala/org/http4s/rho/swagger/models.scala [hunk 1]
```diff
@@ -254,6 +254,17 @@ object models {
       vendorExtensions.foreach { case (key, value) => p.setVendorExtension(key, value) }
       p
     }
+
+    def flatMapOperations(f: Operation => Option[Operation]): Path =
```
Comment (zarthross) [inline swagger/src/main/scala/org/http4s/rho/swagger/models.scala:258]: Not opposed to putting in this code, but naming wise, I think flat map is not appropriate.  This is probably more akin to a `collect` ?

Comment (RafalSumislawski) [reply swagger/src/main/scala/org/http4s/rho/swagger/models.scala:258]: `collestSome` maybe? The signature is more similar to [collestFirstSome](https://github.com/typelevel/cats/blob/a0752346dd0eb2aca54c5a2e153340694d33bada/core/src/main/scala/cats/Foldable.scala#L265) or [collectFoldSome](https://github.com/typelevel/cats/blob/a0752346dd0eb2aca54c5a2e153340694d33bada/core/src/main/scala/cats/Foldable.scala#L333) than to `collect/collectFirst/collectFold` which all take a PartialFunction as an argument.

Comment (zarthross) [reply swagger/src/main/scala/org/http4s/rho/swagger/models.scala:258]: Yeah, `collectSomeOperations` ?  It's wordy but I think captures better than flatMap here.

**Review by zarthross — APPROVED** [review]

**Comment by zarthross** [comment]
Release in https://github.com/http4s/rho/releases/tag/v0.21.0-RC2 should be rolling out to mvn central now

**Pull request merged**